ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/RepositoryBundleContainerAdapterFactory.java

```diff
@@ -0,0 +1,222 @@
+
+		@Override
+		public boolean canEdit(ITargetDefinition target, TreePath treePath) {
+			Object segment = treePath.getLastSegment();
+			return segment instanceof RepositoryBundleContainer || segment instanceof RequirementNode;
+		}
+
+		public boolean canRemove(ITargetDefinition target, TreePath treePath) {
+			return treePath.getLastSegment() instanceof RequirementNode;
+		}
+
+		public IStatus remove(ITargetDefinition target, TreePath[] treePaths) {
+			boolean reload = false;
+			for (TreePath path : treePaths) {
+				Object lastSegment = path.getLastSegment();
+				if (lastSegment instanceof RequirementNode node) {
+					RepositoryBundleContainer container = node.container;
+					RepositoryBundleContainer newContainer = new RepositoryBundleContainer(container.getUri(),
+							container.getRequirements().stream().filter(req -> req != node.requirement).toList());
+					ITargetLocation[] targetLocations = target.getTargetLocations();
+					for (int i = 0; i < targetLocations.length; i++) {
+						ITargetLocation loc = targetLocations[i];
+						if (loc == container) {
+							targetLocations[i] = newContainer;
+						}
+
+					}
+				}
+			}
+			return reload
+					? new Status(IStatus.OK, IPDEConstants.UI_PLUGIN_ID, ITargetLocationHandler.STATUS_FORCE_RELOAD,
+							"reloaded", null) //$NON-NLS-1$
+					: Status.OK_STATUS;
+		}
+	};
+
+	private static final ITreeContentProvider TREE_CONTENT_PROVIDER = new ITreeContentProvider() {
+
+		@Override
+		public boolean hasChildren(Object element) {
+			if (element instanceof RepositoryBundleContainer container) {
+				return !container.getRequirements().isEmpty();
+			}
+			return false;
+		}
+
+		@Override
+		public Object getParent(Object element) {
+			if (element instanceof RequirementNode node) {
+				return node.container;
+			}
+			return null;
+		}
+
+		@Override
+		public Object[] getElements(Object inputElement) {
+			return EMPTY_OBJECTS; // will never be called...
+		}
+
+		@Override
+		public Object[] getChildren(Object parentElement) {
+			if (parentElement instanceof RepositoryBundleContainer container) {
+				return container.getRequirements().stream().map(req -> new RequirementNode(req, container)).toArray();
+			}
+			return EMPTY_OBJECTS;
+		}
+	};
+
+	@Override
+	public <T> T getAdapter(Object adaptableObject, Class<T> adapterType) {
+		if (adaptableObject instanceof RepositoryBundleContainer
+				|| adaptableObject instanceof RequirementNode) {
+			if (adapterType == ILabelProvider.class) {
+				return adapterType.cast(LABEL_PROVIDER);
+			}
+			if (adapterType == ITargetLocationHandler.class) {
+				return adapterType.cast(LOCATION_HANDLER);
+			}
+			if (adapterType == ITreeContentProvider.class) {
+				return adapterType.cast(TREE_CONTENT_PROVIDER);
+			}
+		}
+		return null;
+	}
+
+	@Override
+	public Class<?>[] getAdapterList() {
+		return new Class<?>[] { ILabelProvider.class, ITargetLocationHandler.class, ITreeContentProvider.class,
+				IStyledLabelProvider.class };
```

Comment:
I cannot seem to spot the support for the StyledLabelProvider.

Reply:
Fixed